components/CountryNavigator.js

```diff
@@ -0,0 +1,248 @@
+import { format } from "d3-format";
+import Head from "next/head";
+import { flag } from "country-emoji";
+import Box from "@mui/material/Box";
+import { motion } from "framer-motion";
+import IconButton from "@mui/material/IconButton";
+import Button from "@mui/material/Button";
+import { styled } from "@mui/material/styles";
+import PlayCircleOutlineIcon from "@mui/icons-material/PlayCircleOutline";
+import PauseCircleOutlineIcon from "@mui/icons-material/PauseCircleOutline";
+import ArrowCircleLeftOutlinedIcon from "@mui/icons-material/ArrowCircleLeftOutlined";
+import ArrowCircleRightOutlinedIcon from "@mui/icons-material/ArrowCircleRightOutlined";
+
+import { nth } from "../utils";
+import { RainbowLink } from "../components/RainbowLink";
+import Legend from "../components/Legend";
+import { CATEGORIES_ORDERED_LIST, RAINBOW_COLORS } from "../constants";
+import { getData } from "../data";
+import { useFlagDimensions } from "../hooks/useFlagDimensions";
+import { Flag } from "../components/Flag";
+
+const CountryNavigator = () => {
+  const [year, setYear] = useState(2024);
+  const [sortDict, setSortDict] = useState({
+    type: "ranking",
+    ascending: true,
+  });
+
+  const data = useMemo(
+    () => getData({ years: [year], sortingParams: sortDict }),
+    [year, sortDict]
+  );
+
+  const countries = data.map((d) => d.country);
+  const [currentCountryIndex, setCurrentCountryIndex] = useState(0);
+  const dataCountry = useMemo(() => getData({ countries: countries }), [
+    countries,
+  ]);
+  const [isRunning, setIsRunning] = useState(false);
+
+  useEffect(() => {
+    let interval;
+    if (isRunning) {
+      const updateCountry = () => {
+        setCurrentCountryIndex(
+          (prevIndex) => (prevIndex + 1) % countries.length
+        );
+      };
+
+      interval = setInterval(updateCountry, 1000);
+    }
+
+    return () => clearInterval(interval);
+  }, [isRunning]);
+
+  const currentCountry = countries[currentCountryIndex];
+
+  const currentCountryData = dataCountry.filter(
+    (item) => item.country === currentCountry
+  );
+
+  const years = dataCountry.map((d) => d.year).sort();
+  const uniqueYears = [...new Set(years)];
+
+  const lastYear = [uniqueYears[uniqueYears.length - 1]];
+  const allButLast = uniqueYears.slice(0, uniqueYears.length - 1);
+  const [flagsContainer, flagDimensions] = useFlagDimensions(1.3, 0);
+  const [allFlagsContainer, allFlagDimensions] = useFlagDimensions(
+    allButLast.length,
+    null
+  );
+
+  const handleStart = () => {
+    setIsRunning(true);
+  };
+
+  const handleStop = () => {
+    setIsRunning(false);
+  };
+
+  const handleNextCountry = () => {
+    setCurrentCountryIndex((prevIndex) => (prevIndex + 1) % countries.length);
+  };
+
+  const handlePreviousCountry = () => {
+    setCurrentCountryIndex((prevIndex) =>
+      prevIndex === 0 ? countries.length - 1 : prevIndex - 1
+    );
+  };
+  const SortButton = styled(Button)(({ theme }) => ({
```

Comment:
cont'd on the comment I left on the `<Header>` component 😅  let's extract this `SortButton` to its own component to avoid code duplication